### PR TITLE
Version/select latest release tag

### PR DESCRIPTION
**Pull request by nunoguedelha**
* `YCM_TAG` moved to the latest release `v0.8.1`
* created a cmake file `ProjectsTagsDevel.cmake` for tracking the list of sub-projects to be set in development mode and the branch/tag/commit that should be checkedout when using that mode. This file is not currently used in CMakeLists.txt, but is, for now, meant just for tracking.

Refer to [report in Green-iCub](https://github.com/loc2/lab-organisation/issues/51#issuecomment-401553822) Journal for more information.

### CODE REVIEW

**Comment by nunoguedelha** [comment]
CC @S-Dafarra , @hu-yue , @gabrielenava , @traversaro

File: cmake/ProjectsTagsDevel.cmake [hunk 1]
```diff
@@ -0,0 +1,12 @@
+set(ICUB_TAG devel)
+set(RTF_TAG devel)
+set(WBToolbox_TAG feature/newFactoryAndNewCMakeStructure)
+set(YARP_TAG devel)
+set(codyco-modules_TAG devel)
+set(icub-tests_TAG devel)
+set(iDynTree_TAG devel)
+set(firmware-shared_TAG devel)
```
Comment (traversaro) [inline cmake/ProjectsTagsDevel.cmake]: `icub-` missing ?

Comment (nunoguedelha) [reply cmake/ProjectsTagsDevel.cmake]: > Any reason not to keep WB-Toolbox in devel ?

yes, the selected branch `feature/newFactoryAndNewCMakeStructure` is on top and ahead of the `devel` branch. @diegoferigo confirmed that it has the latest fixes working with the latest version of YARP.
@diegoferigo is something preventing you from merging the branch to `devel`?

Comment (nunoguedelha) [reply cmake/ProjectsTagsDevel.cmake]: > icub- missing ?

sorry I don't understand. The repo name you mention is incomplete..

Comment (traversaro) [reply cmake/ProjectsTagsDevel.cmake]: Sorry, I meant: I guess this line should be `set(icub-firmware-shared_TAG devel)`

Comment (nunoguedelha) [reply cmake/ProjectsTagsDevel.cmake]: ah yes absolutely.

Comment (diegoferigo) [reply cmake/ProjectsTagsDevel.cmake]: @nunoguedelha Not really. Considering the change I applied I was waiting some feedback from the iCub setup but no one complained so far. I will merge it soon.

File: cmake/ProjectsTagsDevel.cmake [hunk 2]
```diff
@@ -0,0 +1,12 @@
+set(ICUB_TAG devel)
+set(RTF_TAG devel)
+set(WBToolbox_TAG feature/newFactoryAndNewCMakeStructure)
```
Comment (diegoferigo) [inline cmake/ProjectsTagsDevel.cmake]: @nunoguedelha You can use devel now

Comment (nunoguedelha) [reply cmake/ProjectsTagsDevel.cmake]: just done and pushed

**Pull request merged**